commcare_connect/opportunity/utils/completed_work.py

```diff
@@ -35,3 +41,16 @@ def update_status(completed_works, opportunity_access, compute_payment=True):
     if compute_payment:
         opportunity_access.payment_accrued = payment_accrued
         opportunity_access.save()
+
+
+def update_work_payment_date(payment: Payment):
```

Comment:
I don't think this logic is quite right, if the worker was paid for less than a full unit, this won't include the carryover. I think you have to iterate over all payments for the user in this function, not the migration, so that if they earned 6000 at one point, but were paid 5000, the 1000 is added to the next payment amount. It at least needs to be all payments that are not already attributed to a work.

The part that calls this in the upload needs to be able to handle that situation as well, it needs to be able to figure out if there is any carryover from previous payments.

Reply:
Thanks for explaining it. [802e47b](https://github.com/dimagi/commcare-connect/pull/405/commits/802e47bfa002fde581ce5067751a2f4e60cc059d)